File: www-cam/src/theme.ts
/** Shared design tokens — single source of truth for all colors, fonts, and spacing. */

export const theme = {
  colors: {
    bg: '#0f1117',
    surface: '#1a1d27',
    border: '#2a2d3a',
    accent: '#4f8cff',
    accentDim: '#2d5299',
    text: '#e0e0e8',
    textDim: '#8888a0',
    danger: '#ff5555',
    success: '#55ff88',
    warning: '#ff9800',
    white: '#ffffff',

    // Port type colors (dataflow)
    portFloat: '#4f8cff',
    portBytes: '#ff9800',
    portText: '#55ff88',
    portSeries: '#ff55aa',
    portAny: '#aaaaaa',

    // Wire colors
    wire: '#4f8cff66',
    wireActive: '#4f8cff',

    // Constraint status colors
    cstFullyConstrained: '#4caf50',
    cstOverConstrained: '#f44336',
    cstUnderConstrained: '#ff9800',
    cstPickHighlight: '#ffeb3b',

    // Simulation colors
    simMaterialRemoval: 'rgba(255,80,80,0.35)',
    simRapid: 'rgba(255,255,100,0.25)',
    simCutting: 'rgba(79,140,255,0.6)',
    simToolShadow: 'rgba(0,0,0,0.3)',
    simToolCutting: 'rgba(255,80,80,0.7)',
    simToolIdle: 'rgba(100,200,100,0.5)',
    simToolOutlineCutting: '#ff5555',
    simToolOutlineIdle: '#55ff88',
    simToolCenter: '#ffffff',

    // CAM preview colors
    camLaser: 'rgba(255, 60, 40, 0.85)',
    camZDefault: '#4f8cff',

    // Hover state
    btnSecondaryHover: '#3a3d4a',

    // Drop zone hover
    dropZoneHoverBg: 'rgba(79,140,255,0.06)',
  },

  fonts: {
    sans: "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif",
    mono: "'Courier New', monospace",
  },

  spacing: {
    sidebarWidth: 320,
    headerHeight: 49,
    pad: 16,
    portRadius: 6,
    portSpacing: 20,
    portOffsetY: 30,
    nodeWidth: 140,
    nodeBaseHeight: 40,
  },
} as const;

/** Return a port color from the theme based on port kind. */
export function portColor(kind: string): string {
  switch (kind) {
    case 'Float': return theme.colors.portFloat;
    case 'Bytes': return theme.colors.portBytes;
    case 'Text': return theme.colors.portText;
    case 'Series': return theme.colors.portSeries;
    default: return theme.colors.portAny;
  }
}

File: hil/combined-frontend/src/theme.rs
// Design tokens -- dark theme matching the TypeScript frontend.

// Colors
pub const BG: &str = "#0a0a0a";
pub const SURFACE: &str = "#1a1a1a";
pub const SURFACE_HOVER: &str = "#252525";
pub const BORDER: &str = "#2a2a2a";
pub const ACCENT: &str = "#3b82f6";
pub const ACCENT_HOVER: &str = "#2563eb";
pub const TEXT: &str = "#e5e7eb";
pub const TEXT_DIM: &str = "#9ca3af";
pub const TEXT_BRIGHT: &str = "#f9fafb";
pub const SUCCESS: &str = "#22c55e";
pub const WARNING: &str = "#f59e0b";
pub const DANGER: &str = "#ef4444";

// Port type colors
pub const PORT_FLOAT: &str = "#22d3ee";
pub const PORT_BYTES: &str = "#f59e0b";
pub const PORT_TEXT: &str = "#4ade80";
pub const PORT_SERIES: &str = "#a78bfa";
pub const PORT_ANY: &str = "#9ca3af";

// Wire colors
pub const WIRE_DEFAULT: &str = "#6b7280";
pub const WIRE_ACTIVE: &str = "#3b82f6";
pub const WIRE_SELECTED: &str = "#f59e0b";

// Spacing (px)
pub const SIDEBAR_WIDTH: f64 = 320.0;
pub const HEADER_HEIGHT: f64 = 49.0;
pub const PORT_RADIUS: f64 = 6.0;
pub const PORT_SPACING: f64 = 20.0;
pub const NODE_BASE_HEIGHT: f64 = 40.0;
pub const NODE_WIDTH: f64 = 190.0;

/// Returns the color for a port kind string.
pub fn port_color(kind: &str) -> &'static str {
    match kind {
        "Float" => PORT_FLOAT,
        "Bytes" => PORT_BYTES,
        "Text" => PORT_TEXT,
        "Series" => PORT_SERIES,
        _ => PORT_ANY,
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_port_color_float() {
        assert_eq!(port_color("Float"), PORT_FLOAT);
    }

    #[test]
    fn test_port_color_bytes() {
        assert_eq!(port_color("Bytes"), PORT_BYTES);
    }

    #[test]
    fn test_port_color_unknown_returns_any() {
        assert_eq!(port_color("Unknown"), PORT_ANY);
    }

    #[test]
    fn test_port_color_all_kinds() {
        assert_eq!(port_color("Text"), PORT_TEXT);
        assert_eq!(port_color("Series"), PORT_SERIES);
        assert_eq!(port_color("Any"), PORT_ANY);
    }
}
